src/block_metadata.rs

```diff
@@ -33,11 +36,22 @@ impl<I: NodeType> Committable for HashableBlock<I> {
     }
 }
 
-#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq, Hash)]
 #[serde(bound = "")]
 pub struct BlockMetadata<I: NodeType> {
-    block_hash: BlockHash<I>,
-    block_size: u64,
-    offered_fee: u64,
-    _phantom: PhantomData<I>,
+    pub block_hash: BuilderCommitment,
+    pub block_size: u64,
+    pub offered_fee: u64,
+    pub signature: <<I as NodeType>::SignatureKey as SignatureKey>::PureAssembledSignatureType,
+    pub _phantom: PhantomData<I>,
```

Comment:
Always put PhantomData last...

Reply:
Is it actually needed here and in `Blockdata` anyways? `I` is already used in other fields

Reply:
`signature` only binds `SignatureKey::PureAssembledSignatureType`, it cannot guarantee that there is a specific unique `I: NodeType` for that binding. Same for `block_payload`. We know it is whatever `NodeType` constrains `BlockPayload` to be, but we don't get the `I` bound to the struct.

We didn't need it when we had `block_hash: BlockHash<I>`, though...